pkg/cluster/delete.go

```diff
@@ -313,6 +313,9 @@ func (m *manager) Delete(ctx context.Context) error {
 		if azureerrors.HasAuthorizationFailedError(err) {
 			err = nil
 		}
+		if azureerrors.ResourceGroupNotFound(err) {
+			err = nil
+		}
```

Comment:
What about error handling on line 310? It looks like an attepmt to handle the same case (`detailedErr.StatusCode == http.StatusNotFound`). Do we still need it?

Reply:
We can se there are attempts to handle the case of an non existing RG in the existing code base  that's true. But I would not remove them for backward compatibility in case this is some change ongoing.

Reply:
Somehow this gives different errors. I suspect some misalignment in azure API. Later might want to dump full error and do more reverse engineering and fix this mess in here :/

Reply:
https://msazure.visualstudio.com/AzureRedHatOpenShift/_workitems/edit/10295993

Reply:
You know I would ask for a work item :)